lib/remote/httpserverconnection.cpp

```diff
@@ -157,9 +157,7 @@ void HttpServerConnection::ProcessMessageAsync(HttpRequest& request)
 		user = ApiUser::GetByName(username);
 
 		/* Deny authentication if 1) given password is empty 2) configured password does not match. */
-		if (password.IsEmpty())
-			user.reset();
-		else if (user && user->GetPassword() != password)
+		if (password.IsEmpty() || !user->ComparePassword(password))
```

Comment:
This will terminate with a SIGSEGV if `user` is not defined. The check should include if (user && ..) and was split for that reason before.

Reply:
Wouldn't `user.reset()` fail as well if user not defined ?